Framework/Algorithms/test/Q1DWeightedTest.h

```diff
@@ -319,6 +319,57 @@ class Q1DWeightedTest : public CxxTest::TestSuite {
     compareWorkspaces(refWedges, outputWedges);
   }
 
+  void testMonochromaticCase() {
+    // Test behaviour when the workspace is monochromatic, and each bin is a different sample and should be kept
+    // separate in the end.
+    LoadNexusProcessed loader;
+
+    loader.initialize();
+    loader.setPropertyValue("Filename", "ILL/D11B/kinetic.nxs");
+    std::string inputWsName = "input";
+    loader.setPropertyValue("OutputWorkspace", inputWsName);
+    loader.execute();
+
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("InputWorkspace", inputWsName))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("OutputWorkspace", "out"))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("OutputBinning", "0,0.002,0.1"))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("NumberOfWedges", "2"))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("WedgeAngle", "90"))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("WedgeWorkspace", "out_wedges"))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setProperty("AccountForGravity", false))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setProperty("ErrorWeighting", false))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setProperty("AsymmetricWedges", false))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("NPixelDivision", "1"))
+
+    TS_ASSERT_THROWS_NOTHING(radial_average.execute())
+    TS_ASSERT(radial_average.isExecuted())
+    MatrixWorkspace_sptr result, wedge1, wedge2;
+    TS_ASSERT_THROWS_NOTHING(
+        result = std::dynamic_pointer_cast<MatrixWorkspace>(AnalysisDataService::Instance().retrieve("out")))
+
+    TS_ASSERT_EQUALS(result->getNumberHistograms(), 3)
+    TS_ASSERT_DELTA(result->y(0)[2], 0.3125, 0.0001)
+    TS_ASSERT_DELTA(result->y(1)[2], 0.3125, 0.0001)
+    TS_ASSERT_DELTA(result->y(2)[2], 0.1875, 0.0001)
+    TS_ASSERT(result->isCommonBins())
+    TS_ASSERT_EQUALS(result->getMaxNumberBins(), 50)
+
+    TS_ASSERT_THROWS_NOTHING(
+        wedge1 = std::dynamic_pointer_cast<MatrixWorkspace>(AnalysisDataService::Instance().retrieve("out_wedges_1")))
+    TS_ASSERT_THROWS_NOTHING(
+        wedge2 = std::dynamic_pointer_cast<MatrixWorkspace>(AnalysisDataService::Instance().retrieve("out_wedges_2")))
+    TS_ASSERT_EQUALS(wedge1->getMaxNumberBins(), wedge2->getMaxNumberBins())
+    TS_ASSERT_EQUALS(wedge1->getNumberHistograms(), wedge2->getNumberHistograms());
```

Comment:
Thanks for extending the test. Could you also assert on `getNumberHistograms` to be 3 and `isCommonBins` also on the wedges? I think the former is the main check to make sure it is not confused with a tof workspace.